shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -321,17 +320,21 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
         new Iterator[InternalRow]() {
           var iter: Iterator[InternalRow] = null
 
-          override def hasNext: Boolean = (iter != null && iter.hasNext) || cbIter.hasNext
+          override def hasNext: Boolean = {
+            // either the current iterator has an element,
+            // else go over the batch and get the next non-degenerate iterator
+            (iter != null && iter.hasNext) ||
+            {
+              while (cbIter.hasNext && !iter.hasNext) {
```

Comment:
I think we can NPE here if `iter` is null and `cbIter.hasNext` is true.  Seems like that would always happen on the first iteration?

Also I found using a block statement in a boolean expression made this harder to follow.